examples/advanced/Tutorial3/reconstruction/FairTestDetectorRecoTask.cxx

```diff
@@ -97,7 +97,9 @@ void FairTestDetectorRecoTask::Exec(Option_t* /*opt*/)
         TVector3 dpos(1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12));
 
         FairTestDetectorHit* hit = new ((*fHitArray)[ipnt]) FairTestDetectorHit(-1, -1, pos, dpos);
-        hit->AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), FairRootManager::Instance()->GetBranchId("FairTestDetectorDigi"), ipnt));
+        if ( FairRootManager::Instance() != NULL ) {
```

Comment:
@karabowi  I agree that this create a problem in this special case. Anyway we have to:

- Find another way to decide if we are in the MQ or ROOT world
- Set the links also in the MQ world.